common/djangoapps/util/db.py

```diff
@@ -82,16 +81,10 @@ def __enter__(self):
             if not self.ALLOW_NESTED and connection.in_atomic_block:
                 raise transaction.TransactionManagementError('Cannot be inside an atomic block.')
 
-            # This will set the transaction isolation level to READ COMMITTED for the next transaction.
-            if self.read_committed is True:
-                if connection.vendor == 'mysql':
-                    cursor = connection.cursor()
-                    cursor.execute("SET TRANSACTION ISOLATION LEVEL READ COMMITTED")
```

Comment:
@natabene Can we have someone from the ops team confirm that the transaction isolation level on prod is the Django default (since Django 2.0): `READ COMMITTED` and also that there is no reason to believe anyone else might have overridden it to anything else?

Reply:
@symbolist Sure, let me check who is available.

Reply:
Hey @symbolist & @jfavellar90.

I don't have all the context on this, but I do remember a few situations which might be relevant.  In the past (around ~9 months ago) edx.org had to ensure we set `tx_issolation` at the database level because some logic was accidentally setting things to `REPEATABLE-READ`, after this situation we both updated the `edx-platform` related code and configured our Aurora 5.7 databases in AWS to enforce [transaction isolation](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_transaction_isolation) to be `READ-COMMITTED` in an attempt to avoid future data inconsistencies.  I think it may be possible that others have overridden this setting in unexpected places, for example by modifying [EDXAPP_MYSQL_OPTIONS](https://github.com/appsembler/configuration/blame/master/playbooks/roles/edxapp/defaults/main.yml#L93)

However, given my understanding I am not concerned with this change, but if we wanted to be extra safe we could explore whether or not we can set the DATABASE -> OPTIONS to `READ-COMMITTED` in the configuration repo.  (I see an example of how to do this for Postgres [here](transaction isolation level on prod is the Django default) which is what leads me to believe we could be doing something similar for MySQL)

Reply:
@adzuci thanks for your reply.
From Django 2.0, `READ-COMMITTED` is the default [isolation level](https://github.com/django/django/pull/7978), however, I think is a good idea your proposal of defining explicitly the database option in the edX configuration repo. Additionally, we can write a comment warning the operators about the consequences of changing that variable (it should be changed only if they know what they're doing). Do you consider this task with the configuration repo should be be a prerequisite to merge this PR? cc @symbolist

Reply:
@adzuci Thanks for the details. Yup, that means this change will be fine on edx.org (and will result in one fewer db query in views which use this decorator).

I think the probability of third-parties using a different isolation level from the default is very low; actually they should not be since it may result in subtle bugs because the code expects a specific isolation level. But that is not my area so to second @jfavellar90's question will you be able to make the call whether to make any changes to configuration? Or maybe just noting in the next release notes that the recommend isolation level is the Django default should be sufficient? 🙂

Reply:
@adzuci I'm in the same line with @symbolist I think we will hardly find cases where the isolation level is different to `READ-COMMITTED` since it could lead to unexpected behaviors and data inconsistencies. I imagine that in most cases, operators are not aware of this setting (as I was 2 months ago before experimenting with ProxySQL). I have a branch [here](https://github.com/edx/configuration/compare/master...eduNEXT:Jhony/edxapp_mysql_isolation_level) for the configuration repo PR. Would it be enough? or better proceed with release notes?

Reply:
Hi @adzuci, waiting for your confirmation if anything needs to be done on the configuration side (e.g. make a note in the release notes). Otherwise this is good to go. 🙂

Reply:
Hey @jfavellar90 and @symbolist, sorry for the slow response.  I think [the configuration PR](https://github.com/edx/configuration/compare/master...eduNEXT:Jhony/edxapp_mysql_isolation_level) is sufficient and don't feel that a release notes entry is needed personally.

I also think you can merge these changes in either order, though I would make sure to get the configuration one merged before the next release.

There is also a chance we may want to set `isolation_level` for other Django service roles in the `configuration` repo, but I think that edxapp is the most important one.

Reply:
@adzuci Thanks!

@jfavellar90 Can you create an PR for the configuration changes branch?